addons/website/static/src/components/wysiwyg_adapter/wysiwyg_adapter.js

```diff
@@ -379,9 +379,35 @@ export class WysiwygAdapterComponent extends Wysiwyg {
             direction: this.websiteService.currentWebsite.metadata.direction,
             showResponsiveFontSizesBadges: true,
             showExtendedTextStylesOptions: true,
+            beforeHistorySystemRemovesElement: this._beforeHistorySystemRemovesElement.bind(this),
+            afterHistorySystemAddsElement: this._afterHistorySystemAddsElement.bind(this),
             ...options,
         });
     }
+    /**
+     * Stops the widgets before an history undo/redo removes a block.
+     *
+     * @private
+     * @param {HTMLElement} el - the element that will be removed
+     */
+    _beforeHistorySystemRemovesElement(el) {
+        this._websiteRootEvent("widgets_stop_request", {
+            editableMode: true,
+            $target: $(el),
+        });
+    }
+    /**
+     * Starts the widgets after an history undo/redo adds a block.
+     *
+     * @private
+     * @param {HTMLElement} el - the element that has been added
+     */
+    _afterHistorySystemAddsElement(el) {
+        this._websiteRootEvent("widgets_start_request", {
```

Comment:
Starting widgets is a async operation. When widgets are started due to an option change, we probably queue that async operation in the editor mutex. Should not it be done here too?

Reply:
@qsm-odoo I did this change :arrow_up: and it is better IMO ! 

The problem is that history repeats itself, as I introduce the change now (which do what [this old commit](https://github.com/odoo/odoo/commit/a153ed42a09f8b7f5e0865112eb7d5affc22a353) did more or less) I introduce the same issue. And so we need something more to fix as you did in [this commit](https://github.com/odoo/odoo/commit/d1a867553da2c99f2895e1ea9ed29f5d6b295a1f) in the past. I think with something similar to what you wanted to do in [this PR](https://github.com/odoo/odoo/pull/74923), we can solve the issue and the the redo/undo flows will be robust.

Just wanted your opinion before moving forward :)